DBG: add support for pretty printing nonzero numbers

Kobzol opened the pull request:
This PR adds support for pretty printing `NonZero` numbers from the standard library. So far I have implemented support for GDB, if you think that my approach is OK, let me know and I'll try to add support for LLDB.

Fixes: https://github.com/intellij-rust/intellij-rust/issues/5247

Kobzol commented:
Rust `1.32.0` doesn't seem to know about `NonZeroI32`, that's why the test is failing. Should I change the test to only include `NonZeroU32`?

Undin commented:
@Kobzol some time ago we introduced special comments to restrict version of rustc that should be used for a particular test
They are `// min-version: %rustc-version%` or `// max-version: %rustc-version%`. You can find them, for example, [here](https://github.com/intellij-rust/intellij-rust/blob/master/pretty_printers_tests/tests/btree.rs#L1)
So you may restrict minimal version of rustc for `NonZero*` test. Or you may split it into several tests and specify min rustc version only for part of them

Kobzol commented:
Thank you, I have added a minimum rustc version comment. When I added tests for LLDB, I have noticed something peculiar: they immediately passed, even though I have not implemented the LLDB formatter for `NonZero` integers yet. As you can see in my latest commit (https://github.com/intellij-rust/intellij-rust/pull/5270/commits/c55a3a7411249ca14de8836bfb21b18a89fa8026), I have purposefully added an error to a different LLDB test and CI failed, as expected, so something is wrong with my setup.

I'm running the tests locally with ` ./gradlew runPrettyPrintersTests --no-daemon`. Do you have any idea what might be causing the problem?

Kobzol commented:
@Undin it seems that LLDB tests are somehow broken in 201. https://travis-ci.org/github/intellij-rust/intellij-rust/builds/678497293 this commit contains invalid test code, yet the LLDB tests passed in 201.

ortem commented:
@Kobzol I'm not sure what might be wrong with your setup. You can try to debug `pretty_printers_tests` by running `pretty_printers_tests/main.rs` with the following command in the run configuration:
```
run --package pretty_printers_test --bin pretty_printers_test -- lldb "<IntelliJ_Rust>/deps/clion-2020.1/bin/lldb/linux/lib/python3.6/site-packages" <193 or 201>
```

Kobzol commented:
Thank you. I didn't notice before that LLDB needs special syntax for checking the output in 201 -- this was the problem. I also used a wrong minimum rustc version. The tests should now be fixed (I hope :) 
 ).

Kobzol commented:
Damn, the tests failed on my fork's Github actions :-( I use the exact same `// min-version: 1.X.Y` annotation that is used in other tests that are successfully skipped.

I thought that this annotation will skip compilation of the test, but it seems that it only skips checking the test debug output, is that correct? In that case I have to modify the test to work with Rust 1.32, otherwise the tests will not pass.

Kobzol commented on prettyPrinters/rust_types.py:
```diff
@@ -48,6 +49,7 @@ class RustType(object):
 STD_REF_REGEX = re.compile(r"^(core::(\w+::)+)Ref<.+>$")
 STD_REF_MUT_REGEX = re.compile(r"^(core::(\w+::)+)RefMut<.+>$")
 STD_REF_CELL_REGEX = re.compile(r"^(core::(\w+::)+)RefCell<.+>$")
+STD_NONZERO_NUMBER_REGEX = re.compile(r"^core::num::NonZero.+$")
```
Thanks!

ortem commented:
@Kobzol Oh, you're right. You can try changing `test_runner.rs` to compile the test after parsing & checking debugger commands (i.e. to call `compile_test` after calling `parse_debugger_commands`).

Kobzol commented:
Thanks, now the tests finally pass. They are possibly also just a teeeeny tiny bit faster now that skipped tests are not being compiled :)

ortem commented:
Thank you!

bors r+
